dev mode; 3.3-M4: dependencies for dev-mode test compilation and maven-compiler-plugin:testCompile differ

While using dev mode in version 3.3-M4 i noticed that dependencies for dev-mode test compilation and maven-compiler-plugin:testCompile differ especially junit (4.12 / 4.13). 

### **How to reproduce:**
1. Add junit in version 4.13 to the classpath in test scope
2. Write a unit test which uses a 4.13 specific method like Assert.assertThrows(...)
3. Run mvn liberty:dev

As result you get:

> [INFO] ************************************************************************
[INFO] *    Liberty is running in dev mode.
[INFO] *        To run tests on demand, press Enter.
[INFO] *        To restart the server, type 'r' and press Enter.
[INFO] *        To stop the server and quit dev mode, press Ctrl-C or type 'q' and press Enter.
[INFO] *        Liberty server HTTP port: 12000
[INFO] *        Liberty server HTTPS port: 12001
[INFO] *        Liberty debug port: 12002
[INFO] ************************************************************************
[INFO] Source compilation was successful.
D:\repos\example-service\de.markant.messe.example.service\src\test\java\de\markant\messe\example\service\ExampleServiceImplTest.java:4: **error: cannot find symbol**
import static **org.junit.Assert.assertThrows;**
^
  symbol:   static assertThrows
  location: class Assert
D:\repos\example-service\de.markant.messe.example.service\src\test\java\de\markant\messe\example\service\ExampleServiceImplTest.java:50: **error: cannot find symbol**
    	**assertThrows**(IllegalCallerException.class, () -> service.throwExceptionsForTest());
    	^
  symbol:   method assertThrows(Class<IllegalCallerException>,()->servic[...]est())
  location: class ExampleServiceImplTest
.....
[INFO] Tests compilation had errors.

After some research and testing (elimination of transitive junit dependencies, running alot of debug with .m2/junit purging 
 etc.)  i noticed that the only place where junit:4.12 is coming in, is through liberty-maven-plugin itself. 
dev-mode seams to ignore junit:4.13 from pom.xml definition.

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java
@@ -190,6 +190,9 @@ public class StartDebugMojoSupport extends BasicSupport {
     protected Plugin getPlugin(String groupId, String artifactId) {
         Plugin plugin = project.getPlugin(groupId + ":" + artifactId);
         if (plugin == null) {
+            plugin = getPluginFromPluginManagement(groupId, artifactId);
+        }
+        if (plugin == null) {
             plugin = plugin(groupId(groupId), artifactId(artifactId), version("RELEASE"));
         }
         return plugin;

